client/my-sites/domains/domain-management/list/site-domains.jsx

```diff
@@ -464,51 +480,27 @@ export class SiteDomains extends Component {
 	};
 }
 
-const changePrimary = ( domain, mode ) =>
-	composeAnalytics(
-		recordGoogleEvent(
-			'Domain Management',
-			'Changed Primary Domain to in List',
-			'Domain Name',
-			domain.name
-		),
-		recordTracksEvent( 'calypso_domain_management_list_change_primary_domain_click', {
-			section: domain.type,
-			mode,
-		} )
-	);
-
-export default connect(
-	( state, ownProps ) => {
-		const siteId = ownProps?.selectedSite?.ID || null;
-		const userCanManageOptions = canCurrentUser( state, siteId, 'manage_options' );
-		const selectedSite = ownProps?.selectedSite || null;
-		const isOnFreePlan = selectedSite?.plan?.is_free || false;
-		const purchases = getPurchases( state );
-		const productsList = getProductsList( state );
-
-		return {
-			currentRoute: getCurrentRoute( state ),
-			hasDomainCredit: !! ownProps.selectedSite && hasDomainCredit( state, siteId ),
-			hasProductsList: 0 < ( Object.getOwnPropertyNames( productsList )?.length ?? 0 ),
-			isDomainOnly: isDomainOnlySite( state, siteId ),
-			isAtomicSite: isSiteAutomatedTransfer( state, siteId ),
-			hasNonPrimaryDomainsFlag: getCurrentUser( state )
-				? currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
-				: false,
-			isOnFreePlan,
-			userCanManageOptions,
-			canSetPrimaryDomain: siteHasFeature( state, siteId, FEATURE_SET_PRIMARY_CUSTOM_DOMAIN ),
-			purchases,
-			isFetchingPurchases: isFetchingSitePurchases( state ),
-		};
-	},
-	{
-		changePrimary,
```

Comment:
Should we really abandon the auto-binding to `dispatch`? Couldn't we at least manually provide the binding here, and keep the dispatching syntax of the thunks the same way as before? We could also provide `dispatch` to the list of returned props as necessary.